src/ledger/SorobanMetrics.cpp

```diff
@@ -0,0 +1,187 @@
+#include "ledger/SorobanMetrics.h"
+
+#include "medida/meter.h"
+#include "medida/metrics_registry.h"
+#include "medida/timer.h"
+
+namespace stellar
+{
+SorobanMetrics::SorobanMetrics(medida::MetricsRegistry& metrics)
+    : /* ledger-wide metrics */
+    mLedgerTxCount(metrics.NewHistogram({"soroban", "ledger", "tx-count"}))
```

Comment:
This design is a bit error-prone since all the metrics are accessible by every user. I think we can merge this as is, but it would be useful to group metrics by operation/use case and pass the respective group to the respective users. Basically it would be nice to have the structure of this class to reflect the comments. Again, that's not for this PR.

Reply:
Yeah if we adding more metrics (which we probably will), it's worthwhile to split it into sub structures as it's getting bloated.